File: paperworks-api/src/main/java/com/amaurote/controller/catalogue/AuthorController.java
package com.amaurote.controller.catalogue;

import com.amaurote.catalogue.service.AuthorService;
import com.amaurote.controller.BaseController;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/catalogue")
public record AuthorController(AuthorService service) implements BaseController {
}

File: paperworks-api/src/main/java/com/amaurote/controller/catalog/AuthorController.kt
package com.amaurote.controller.catalog

import com.amaurote.catalog.service.AuthorService
import com.amaurote.controller.BaseControllerKT
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/catalog")
class AuthorController(private val authorService: AuthorService) : BaseControllerKT() {
}